src/xmipp/libraries/data/grids.h

```diff
@@ -183,6 +183,11 @@ class SimpleGrid
         \\ Ex: SimpleGrid sg2(sg1); */
     SimpleGrid(const SimpleGrid &SG);
 
+    /** Destructor.
+     * Explicitly defined to avoid Sonarcloud complains
+     */
+    ~SimpleGrid() {}
```

Comment:
Not really. The empty constructor is not trivial. Not removed.

Reply:
It is OK that the non-parametric constructor is non-trivial.
My argument is that you don't need to manually define destructor, copy constructor and assignment operator, because you can follow the rule of Zero.

Reply:
OK, I have removed them.